src/lib/sql/sql_translator.cpp

```diff
@@ -466,7 +497,27 @@ SQLTranslator::TableSourceState SQLTranslator::_translate_table_origin(const hsq
 
   switch (hsql_table_ref.type) {
     case hsql::kTableName: {
-      if (StorageManager::get().has_table(hsql_table_ref.name)) {
+      // WITH descriptions or sub-queries are treated as though they were inline views or tables
+      // They mask existing tables or views with the same name.
+      if (_with_descriptions.count(hsql_table_ref.name)) {
+        const auto iter = _with_descriptions.find(hsql_table_ref.name);
```

Comment:
count() searches, find() searches - a single find() should be sufficient. Also, names iterators I like better, e.g. "with_descriptions_iterator"

Reply:
Good catch!

Reply:
This PR looks like the comments have been seen, but not addressed yet. Skipping the rest of the comments and my own review for now.

Reply:
Correct, I just started addressing them.